From 4672961d01004443b81aa710fadeb6eadbf39488 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Thu, 21 Aug 2025 18:28:02 -0700
Subject: [PATCH 21/21] Tensile disable hwmonitor for gfx115x

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 shared/tensile/Tensile/Common.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shared/tensile/Tensile/Common.py b/shared/tensile/Tensile/Common.py
index 5336a3ac8a..524dd1e2d4 100644
--- a/shared/tensile/Tensile/Common.py
+++ b/shared/tensile/Tensile/Common.py
@@ -2464,7 +2464,7 @@ def assignGlobalParameters( config, capabilitiesCache: Optional[dict] = None ):
     if os.name == "nt":
       globalParameters["CurrentISA"] = (9,0,6)
       printWarning("Failed to detect ISA so forcing (gfx906) on windows")
-  isasWithDisabledHWMonitor = ((9,4,2), (9,5,0), (11,0,0), (11,0,1), (11,0,2), (12,0,0), (12,0,1))
+  isasWithDisabledHWMonitor = ((9,4,2), (9,5,0), (11,0,0), (11,0,1), (11,0,2), (11,0,50), (11,0,51), (12,0,0), (12,0,1))
   if globalParameters["CurrentISA"] in isasWithDisabledHWMonitor:
     isaString = ', '.join(map(gfxName, isasWithDisabledHWMonitor))
     printWarning(f"HardwareMonitor currently disabled for {isaString}")
-- 
2.43.0

